index.js

```diff
@@ -957,7 +957,8 @@ async function relatedDiscussions (req, res) {
               parentCid: payload.pid,
               name: payload.name || constants.defaultCategory
             }
-            const cdata = await getResponseData(req, constants.createCategory, createRelatedDiscussions, body, constants.post);
+            const categoryUrl = `${constants.createCategory}?_uid=${payload.uid}`
```

Comment:
If uid is not exist in the request then it will not work. Your fallback logic should work.